jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -124,6 +125,15 @@ def _parse_image(app, file_obj, data_label, ext=None):
     data_iter = get_image_data_iterator(app, file_obj, data_label, ext=ext)
 
     for data, data_label in data_iter:
+        if data.coords is not None:
+            if isinstance(data.coords, GWCS):
+                # keep a copy of the original bounding box so we can detect
+                # when extrapolating beyond, but then remove the bounding box
+                # so that image layers are not cropped.
+                # NOTE: if extending this beyond GWCS, the mouseover logic
+                # for within_bounding_box should also be updated
+                data.coords._orig_bounding_box = getattr(data.coords, 'bounding_box', None)
+                data.coords.bounding_box = None
```

Comment:
Actually, I am not sure about this approach anymore. While writing a test for this, I realize that this actually modifies the input, which can lead to surprising behavior in analysis. Consider this use case:

1. User creates the GWCS by hand.
2. User pass this object into NDData to be loaded by Imviz.
3. User then uses this GWCS in the notebook for something entirely different. Now, bounding box is silently set to `None` by Imviz parser, but user doesn't know. User thinks out-of-bounds transformation is valid.

Reply:
Ok, what if we restore the bounding box in `get_data_from_viewer`?

Reply:
I am digging at GWCS API and see if there is something we can use that doesn't involve changing the actual bounding_box attribute. Will report back.

Reply:
Alas, I am not having a lot of luck. https://github.com/spacetelescope/gwcs/issues/431

Reply:
I'm not as concerned about the modifying input as it is _importing_ data and not _necessarily_ expected to fully roundtrip, but if we can come up with a cleaner way to retain that information, that would definitely be useful.  In other words, I think the bug/annoyance/loss of functionality resulting from keeping the bounding box is worse than dropping it during import.  We could even raise a snackbar or logger warning indicating that it was dropped.

Reply:
I agree. I did find a bug, so I'll push commit to this PR shortly.